### ISSUE TITLE
508-defect-2 [COGNITION]: A heading SHOULD exist within the sponsor information accordion

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#16763]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:**  Josh Kim

## User Story or Problem Statement

As a screen reader user, I want to navigate the hierarchy of the page content using heading levels to save time and frustration.

## Details

Two heading-related issues exist on this page:

- Accordion titles are not coded as headings.
- No heading is present for the sponsor information accordion which breaks the pattern of headings on the page as every applicant information page has a heading.

This results in the second accordion for the sponsor's information to semantically disappear from the page as none of its contents contain a heading. For screen reader users, this is a serious issue as portions of the page may not be intuitively navigable.

## Acceptance Criteria

- [ ] Accordion titles are coded as `h3` headings.
- [ ] Headings within the accordion are coded as `h4` headings.
- [ ] A heading is added within the sponsor information accordion.

## Environment

* Operating System: all
* Browser: any
* Screenreading device: any
* Server destination: staging, production

## Steps to Recreate

1. Complete Form 28-8832 as a child or spouse.
2. On step 3 of 3: review application, expand the second accordion "Your sponsoring Veteran or service member."
3. Verify that no heading exists next to edit.
4. Open up dev tools and inspect the accordion button.
5. Verify that no heading exists.

## Solution, Accordion Headings

Wrap the button with a `h3` and convert the following `h3`s within the accordion to `h4`s.

### Before

```html
<!-- Accordion heading -->
<div class="accordion-header clearfix schemaform-chapter-accordion-header">
  <button class="usa-button-unstyled" aria-expanded="true" aria-controls="collapsible-11">
    Applicant Information
  </button>
</div>

<!-- Accordion contents -->
<h3 class="vads-u-font-size--h5 vads-u-margin--0">
  Claimant Information
</h3>

```

### After

``` diff
<!-- Accordion heading -->
+ <h3 class="vads-u-font-size--h4">
   <div class="accordion-header clearfix schemaform-chapter-accordion-header">
     <button class="usa-button-unstyled" aria-expanded="true" aria-controls="collapsible-11">
       Applicant Information
     </button>
   </div>
+ </h3>
<!-- Repeat change for all -->

<!-- Accordion contents -->
+ <h4 class="vads-u-font-size--h5 vads-u-margin--0">
     Claimant Information
+ </h4>
<!-- Repeat change for all -->
```

## Solution, Missing Heading in Sponsor Information

Simply add a heading to the div that contains the edit button.

### Before

```html
<div class="form-review-panel-page-header-row">
  <button type="button" class="edit-btn primary-outline" aria-label="Edit Your sponsoring Veteran or service member">
    Edit
  </button>
</div>
```

### After

``` diff
<div class="form-review-panel-page-header-row">
+  <h4 class="vads-u-font-size--h5 vads-u-margin--0">
+    Sponsor Information
+  </h4>
  <button type="button" class="edit-btn primary-outline" aria-label="Edit Your sponsoring Veteran or service member">
    Edit
  </button>
</div>
```

## WCAG or Vendor Guidance 

* [Headings - Web Accessibility Tutorials](https://www.w3.org/WAI/tutorials/page-structure/headings/)
* [WCAG 2.0 1.3.1 Info and Relationships A](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships)

### Before
<img width="617" alt="Screen Shot 2020-12-02 at 10 44 21 AM" src="https://user-images.githubusercontent.com/14154792/100895531-aba8d880-348b-11eb-9a38-67243ab43209.png">

### After
For this example, I used the `h4` utility class to style the `h3` accordion title to look like a `h4` to match the visual order on the page as its children uses a `h5` utility class.

<img width="1447" alt="Screen Shot 2020-12-02 at 10 58 57 AM" src="https://user-images.githubusercontent.com/14154792/100897150-684f6980-348d-11eb-80ba-8ffbcd4d9409.png">

**Comment by jason-gcio** [comment]
Super easy, lets do it.

**Comment by micahchiang** [comment]
fix included in https://github.com/department-of-veterans-affairs/vets-website/pull/15507
